crates/cranelift/src/func_environ.rs

```diff
@@ -2334,18 +2335,23 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
         let mut values = vec![];
         if valtypes.len() == 0 {
             // OK
-        } else if valtypes.len() == 1 {
-            let offset = i32::try_from(self.offsets.vmctx_typed_continuations_payloads()).unwrap();
-            let val = builder
-                .ins()
-                .load(convert_type(valtypes[0]), memflags, base_addr, offset);
-            values.push(val)
+        } else if valtypes.len() <= MAXIMUM_CONTINUATION_PAYLOAD_COUNT as usize {
+            let mut offset =
+                i32::try_from(self.offsets.vmctx_typed_continuations_payloads()).unwrap();
+            for valtype in valtypes {
+                let val = builder
+                    .ins()
+                    .load(convert_type(*valtype), memflags, base_addr, offset);
```

Comment:
I am positive there is already logic somewhere for converting types... We should try to find it. At least I know there is logic for determining the size of types.

Reply:
found it